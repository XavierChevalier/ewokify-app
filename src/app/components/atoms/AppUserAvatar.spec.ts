import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import { PicsumPhoto } from './AppUserAvatar.stories'
import AppUserAvatar from './AppUserAvatar.vue'
import { ComponentTestsGenerator } from '@/tests/tests-generators/ComponentTestsGenerator'

describe('AppUserAvatar', () => {
  generateStorybookSnapshotTests({ PicsumPhoto })

  const componentTestsGenerator = new ComponentTestsGenerator(AppUserAvatar)
  componentTestsGenerator.itShouldBeDefined()

  componentTestsGenerator
    .property<string>('url')
    .itShouldBeDefined()
    .itShouldHaveAValidatorFunction()
    .itShouldSuccessWith('https://picsum.photos/50')
    .itShouldFailWith('empty string', '')
    .itShouldFailWith('null', null)

  componentTestsGenerator
    .property<string>('size')
    .itShouldBeDefined()
    .itShouldHaveADefaultValue()
    .itShouldHaveAValidatorFunction()
    .itShouldSuccessWith('sm')
    .itShouldSuccessWith('base')
    .itShouldSuccessWith('lg')
    .itShouldFailWith('unknown size type', 'unknown')

  componentTestsGenerator.generateTests()
})
