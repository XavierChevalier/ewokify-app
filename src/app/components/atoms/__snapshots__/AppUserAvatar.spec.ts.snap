// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppUserAvatar [snapshot] should render the "PicsumPhoto" component correctly 1`] = `
<img
  alt="The TheBaronless's avatar"
  class="box-content p-0.5 rounded-full border-2 border-neutral-light w-10 h-10"
  src="https://picsum.photos/50"
/>
`;
